pytravharv/pytravharv.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import logging.config
+import os
+from typing import List, Optional
+
+import yaml
+from pyrdfstore import create_rdf_store
+from rdflib import Graph
+
+from pytravharv.common import QUERY_BUILDER, insert_resource_into_graph
+from pytravharv.store import TargetStoreAccess as RDFStoreAccess
+from pytravharv.TravHarvConfigBuilder import (
+    TravHarvConfig,
+    TravHarvConfigBuilder,
+)
+from pytravharv.TravHarvExecuter import TravHarvExecutor
+
+# log = logging.getLogger("pyTravHarv")
+log = logging.getLogger(__name__)
+
+
+class TravHarv:
+    """Assert all paths for given subjects.
+    Given a configuration file, assert all paths for all subjects in the configuration file.
+    :param config_folder: str
+    :param name: str
+    :param output: str
+    :param context: list[str]
+    :param target_store_info: Optional[list[str]]
+    :param verbose: bool
+    """
+
+    def __init__(
+        self,
+        config_folder: str,
+        name: Optional[str] = None,
+        output: Optional[str] = None,
+        context: Optional[List[str]] = None,
+        target_store_info: Optional[List[str]] = None,
+        verbose: bool = True,
+    ):
+        """Assert all paths for given subjects.
+        Given a configuration file, assert all paths for all subjects in the configuration file.
+        :param config_folder: str
+        :param name: str
+        :param output: str
+        :param context: list[str]
+        :param target_store: list[str]
+        :param verbose: bool
+        """
+        self.config_folder = config_folder
+        self.name = name
+        self.output = output
+        self.context = context  # have better names for this
+
+        if verbose:
+            file_location = os.path.dirname(os.path.realpath(__file__))
+            with open(
+                os.path.join(file_location, "debug-logconf.yml"), "r"
+            ) as f:
+                config = yaml.safe_load(f.read())
+                logging.config.dictConfig(config)
+
+            self.target_store = create_rdf_store(*target_store_info)
+        self.target_store_access = RDFStoreAccess(
+            self.target_store, QUERY_BUILDER
+        )
+
+        # if there is context add it to the target store
+        if context is not None:
```

Comment:
Something indeed to discuss but I would like to keep this in the travharv package since this was also meant to be a standalone package and preloading triples for selection can be essential.

Reply:
issue mentioned in #19